### ISSUE TITLE
no-shadowed-variable complains about “shadowing” a type by a variable

### ISSUE THREAD

**Issue by m93a** [palantir/tslint#4247]
### Bug Report

- __TSLint version__: 5.11.0
- __TypeScript version__: 3.3
- __Running TSLint via__: VSCode

#### TypeScript code being linted

```ts
type foo = 0;
function bar(a: foo)
{
    let foo = '';
}
```

with `tslint.json` configuration:

```json
{
  "rules": {
    "no-shadowed-variable": true
  }
}
```

#### Actual behavior
`[tslint] Shadowed name: 'foo'`

#### Expected behavior
TSLint shouldn't throw an error for declaration merging, even accross scopes. Both the type `foo` and the variable `foo` are completely usable and do not shadow each other.

**Comment by JoshuaKGoldberg** [comment]
> are completely usable and do not shadow each other.

Beyond removing ambiguities at runtime, the rule also helps by making it visually clear for developers what refers to what (reduced cognitive overhead). Maybe allowing shadowed names in this manner should be a rule option?

**Comment by m93a** [comment]
Yes, allowing this as an option would certainly be the best solution.

But not having such an option seems quite limiting to me, as I often use the same name for a type and an argument name of that type – like this:
```typescript
type renderer<T> = (value: T) => React.ReactNode;

function renderNode<T>(renderer: renderer<T>)
{
  return renderer( something() );
}
```

**Comment by JoshuaKGoldberg** [comment]
☠️ *TSLint's time has come!* ☠️ 

TSLint is no longer accepting most feature requests per #4534. See **[typescript-eslint.io](https://typescript-eslint.io)** for the new, shiny way to lint your TypeScript code with **ESLint**. ✨ 

It was a pleasure open sourcing with you all!

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
